src/prepare.js

```diff
@@ -1,16 +1,31 @@
 import { readFile, writeFile } from "fs/promises";
 import { getPluginFiles } from "./util.js";
+import isPrerelease from "./is-prerelease.js";
 
 export async function prepare(_, context) {
     const version = context.nextRelease.version;
     const fileMap = await loadFileMap();
     const { minAppVersion } = fileMap.get("manifest.json");
 
+    // TODO(TfTHacker/obsidian42-brat#93): Remove if BRAT updates to GitHub release based workings
+    // Pre-release, add `manifest-beta.json` key to filemap
+    if (isPrerelease(context.branch)) {
+        fileMap.set("manifest-beta.json", {
+            ...fileMap.get("manifest.json"),
+        });
+    }
+
     fileMap.forEach((json, file) => {
         switch (file) {
             case "versions.json":
                 json[version] = minAppVersion;
                 break;
+            case "manifest.json":
+                // Don't touch manifest.json in case of pre-release
+                if (!isPrerelease(context.branch)) {
+                    json.version = version;
+                }
+                break;
```

Comment:
Shouldn't we still bump `manifest-beta.json`'s version so that BRAT detects that a new version is available? Otherwise it seems like it'll never "see" the new versions since it's always copying from `manifest.json`, which isn't being changed.

Reply:
Unless I'm completely missing something, this should already be happening: if we work on a pre-release, we push `manifest-beta.json` to the map as a copy of `manifest.json` before we run the code to update the version in `manifest-beta.json`. If we're on a normal release, we won't touch `manifest-beta.json` but update `manifest.json` instead.

Reply:
Ah right, my own code confused me 😎 I forgot that the default case statement will handle it. Please disregard!